CONTRIBUTING.md

```diff
@@ -71,6 +75,72 @@ IntelliJ's code generation can be configured to do this automatically:
      Tick:
      - Make generated local variables final
      - Make generated parameters final
+     
+**Null checking of method or function parameters**
+
+If arguments to a constructor set members of the object, or if arguments to a method
+change internal state, e.g. adding an item to a list, they should be checked for null, e.g.
+
+```
+public class MyObject {
+
+    private final String name;
+    private final List<String> myList = new ArrayList<>();
+
+    public MyObject(final String name) {
+        this.name = Objects.requireNonNull(name, "name");
+    }
+
+    public void addItem(final String item) {
+        myList.add(Objects.requireNotNull(item, "item"));
+    }
+}
+
+```
+
+**No TODOs in code**
+
+Please don't leave any TODOs in the code
+
+**New line for each method or constructor parameter**
+
+Methods or constructors should be defined with each parameter on a new line.
+
+```
+public void doSomething(
+    final String name,
+    final int age,
+    final boolean human
+) {
+    // body of method
+}
+
+```
+
+**Private static methods**
+
+Private methods that don't use object state should be declared as static.
```

Comment:
I'm not a fan either. Imho, whether it's static or not makes little difference and doesn't justify an exception from the standard rules.

Reply:
This comes down to code readability. 

If a method is `static` then its function name is displayed in italics in the IDE. This means you can read the code that calls the method and _know_ that its static and hence has no way of accessing or mutating instance fields.

I bet your IDE is even reporting methods that could be static but aren't as a warning. (unless you've turned it off). 

Of course, there are times where it doesn't make sense to make a public method static, e.g. you want non-static as it's being used in dependency injection.  That's obviously totally valid. In such an instance I disable the warning and add a comment:

```java
@SuppressWarnings("MethodMayBeStatic")
public Thing createThing() {
}
```

Reply:
Exactly why I think it's context dependent - if I'm writing a method and I want to communicate to future developers that it is safe to use without worrying about side effects on the class variables I will label it static. That being said, just because it doesn't change any variables today doesn't mean I want to communicate that guarantee (I reserve the right to change that)! I've also seen this backfire when people pass the instance of the class (as `this`) to a static method, which defeats the whole purpose of it. 

Usually, I'll write methods as static if possible, but I don't think this should be a rule.

Reply:
Since this is only for private methods, why wouldn't you want to communicate the strongest guarantee possible at the time?

Reply:
exactly.